alarm/PolicyManager2.js

```diff
@@ -2106,45 +2106,69 @@ class PolicyManager2 {
 
   async batchPolicy(actions) {
     let results = {
-      'create':[],
-      'update':[],
-      'delete':[]
+      'create': [],
+      'update': [],
+      'delete': [],
+      'disable': [],
+      'enable': [],
     };
-    for(const action in actions){
+    for (const action in actions) {
       const rawData = actions[action] || [];
       switch (action) {
         case 'create':
-          for(const rawPolicy of rawData){
+          for (const rawPolicy of rawData) {
             const { policy, alreadyExists } = await this.checkAndSaveAsync(new Policy(rawPolicy));
-            let result = policy;
-            if (alreadyExists == 'duplicated') {
-              result = 'duplicated'
-            }
-            results.create.push(result);
+            results[action].push({
+              policy: policy,
+              error: alreadyExists == 'duplicated'
+            });
           }
           break;
         case 'update':
-          for(const rawPolicy of rawData){
+          for (const rawPolicy of rawData) {
             const pid = rawPolicy.pid;
-            const oldPolicy = await this.getPolicy(pid);
-            await this.updatePolicyAsync(rawPolicy);
-            const newPolicy = await this.getPolicy(pid);
-            this.tryPolicyEnforcement(newPolicy, 'reenforce', oldPolicy);
-            results.update.push(newPolicy);
+            let newPolicy;
+            try {
+              const oldPolicy = await this.getPolicy(pid);
+              await this.updatePolicyAsync(rawPolicy);
+              newPolicy = await this.getPolicy(pid);
+              this.tryPolicyEnforcement(newPolicy, 'reenforce', oldPolicy);
+            } catch (e) {
+              log.warn(`update policy ${rawPolicy} error`,e);
+            }
+            results[action].push({
+              policy: newPolicy,
+              error: !newPolicy
+            });
           }
           break;
         case 'delete':
-          for(const policyID of rawData){
+          for (const policyID of rawData) {
             let policy = await this.getPolicy(policyID);
-            let result;
             if (policy) {
               await this.disableAndDeletePolicy(policyID);
               policy.deleted = true;
-              result = policy;
             } else {
-              result = "invalid policy";
+              log.warn(`invalid policy ${policyID}`);
+            }
+            results[action].push({
+              policy: policy,
+              error: !policy
+            });
+          }
+          break;
+        case 'enable':
+        case 'disable':
+          const pmAction = action == 'enable' ? this.enablePolicy : this.disablePolicy;
```

Comment:
should we consider combine enable/disable into update? this feels like extra complexity

Reply:
enable and disable may only pass policy ID